src/openassetio-python/tests/package/hostApi/test_manager.py

```diff
@@ -1916,6 +1920,619 @@ def test_when_called_with_None_data_then_TypeError_is_raised(
+
+        with pytest.raises(expected_exception, match=batch_element_error.message) as exc:
+            manager.register(some_refs, some_entity_traitsdatas, a_context)
+
+        method.assert_called_once_with(
+            some_refs,
+            some_entity_traitsdatas,
+            a_context,
+            a_host_session,
+            mock.ANY,
+            mock.ANY,
+        )
+
+        assert exc.value.index == expected_index
+        assert_BatchElementError_eq(exc.value.error, batch_element_error)
+
+
+class Test_Manager_register_with_batch_throwing_overload:
+    def test_when_success_then_multiple_EntityReferences_returned(
+        self,
+        manager,
+        mock_manager_interface,
+        a_host_session,
+        some_refs,
+        some_entity_traitsdatas,
+        a_context,
+        some_different_refs,
+    ):
+        method = mock_manager_interface.mock.register
+
+        def call_callbacks(*_args):
+            # Success
+            callback = method.call_args[0][4]
+            callback(0, some_different_refs[0])
+
+            callback = method.call_args[0][4]
+            callback(1, some_different_refs[1])
+
+        method.side_effect = call_callbacks
+
+        actual_refs = manager.register(
+            some_refs,
+            some_entity_traitsdatas,
+            a_context,
+            Manager.BatchElementErrorPolicyTag.kException,
+        )
+
+        method.assert_called_once_with(
+            some_refs,
+            some_entity_traitsdatas,
+            a_context,
+            a_host_session,
+            mock.ANY,
+            mock.ANY,
+        )
+
+        assert len(actual_refs) == 2
+        assert actual_refs[0] == some_different_refs[0]
+        assert actual_refs[1] == some_different_refs[1]
+
+    def test_when_success_out_of_order_then_EntityReferences_returned_in_order(
+        self,
+        manager,
+        mock_manager_interface,
+        a_host_session,
+        some_refs,
+        some_entity_traitsdatas,
+        a_context,
+        some_different_refs,
+    ):
+        method = mock_manager_interface.mock.register
+
+        def call_callbacks(*_args):
+            # Success
+            callback = method.call_args[0][4]
+            callback(1, some_different_refs[1])
+
+            callback = method.call_args[0][4]
+            callback(0, some_different_refs[0])
+
+        method.side_effect = call_callbacks
+
+        actual_refs = manager.register(
+            some_refs,
+            some_entity_traitsdatas,
+            a_context,
+            Manager.BatchElementErrorPolicyTag.kException,
+        )
+
+        method.assert_called_once_with(
+            some_refs,
+            some_entity_traitsdatas,
+            a_context,
+            a_host_session,
+            mock.ANY,
+            mock.ANY,
+        )
+
+        assert len(actual_refs) == 2
+        assert actual_refs[0] == some_different_refs[0]
+        assert actual_refs[1] == some_different_refs[1]
```

Comment:
[fix] Any reason not to be using the new approach here to save having to revisit this later?

Reply:
No good reason other than consistency until #886. Will fix as part of this PR.